atlp-rwanda/e-commerce-bitcrafters-bn
# [#187364726] -On successful registration, users should receive a JWT

**What does this PR do?**
 This pull request adds a feature for users to sign up and get a special key to use our website. It also makes sure users get clear messages if they make a mistake when signing up.

 **Description of Task to be completed?**

- Made it possible for users to sign up (POST /users/signup).
- Gave users a special key (JWT) when they sign up.
- Checked if the information users give when signing up is correct and gave them helpful messages if it's not.
- Also there is new database configuration by using online database for easy deployment and testing.

 **How should this be manually tested?**

1. Try to use Postman or ThunderClient to test this endpoints
2. Try signing up with correct information and see if you get a special key back.
3. Try signing up with wrong information and check if you get clear messages about what went wrong.

**Any background context you want to provide?**
This makes our website safer and easier to use. Users can now sign up and get a special key to use our services securely.

**What are the relevant pivotal tracker/Trello stories?**
  [#187364726]
